### PR TITLE
backend/feat: #769 version label for middleware metrics

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
https://github.com/nammayatri/nammayatri/issues/769

To do staggered deployments, we need to compare 4xx/ 5xx b/w the current and previous version across routes.
To monitor through Prom metrics, we need to add the version label in the metric: http_request_duration_seconds_count.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## How did you test it?
Tested locally. Middleware metric `http_request_duration_seconds` worked the same as in master branch, only `version` label added

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -27,6 +30,8 @@
           source-overrides = {
             passetto-client = inputs.passetto-hs + /client;
             passetto-core = inputs.passetto-hs + /core;
+            wai-middleware-prometheus = inputs.wai-middleware-prometheus + /wai-middleware-prometheus;
+            prometheus-client = inputs.wai-middleware-prometheus + /prometheus-client;
```
Comment (srid) [inline flake.nix:34]: You want to use https://github.com/juspay/prometheus-haskell/pull/3 and use prometheus exactly like `euler-hs` is being used, ie., via `imports` and `haskellFlakeProjectModules`.

https://github.com/nammayatri/shared-kernel/blob/12e660da6fcb5d09dc5a7b05b593a44c9583ee03/flake.nix#L24-L26

@aravindgopall @paulvictor Can we merge that PR?

**Review by srid — CHANGES_REQUESTED** [review]
See https://github.com/nammayatri/shared-kernel/pull/112/files#r1189810733

Let's collaborate with @aravindgopall to get that PR merged, and you can make your contributions on top of it.

File: flake.nix [hunk 2]
```diff
@@ -9,6 +9,9 @@
     passetto-hs.flake = false;
     clickhouse-haskell.url = "github:piyushKumar-1/clickhouse-haskell";
     clickhouse-haskell.flake = false;
+    # wai-middleware-prometheus.url = "/home/roman/Projects/beckn/nammayatri/prometheus-juspay/prometheus-haskell";
```
Comment (NikithShetty) [inline flake.nix:12]: Needs to be updated once the PR gets merged

File: flake.nix (hunk 1 above)
Comment (srid) [reply flake.nix:34]: The PR is merged. @roman-bodavskiy You can just use https://github.com/juspay/prometheus-haskell/tree/more-proc-metrics like we use `euler-hs` here. No need to add any of these `source-overrides`.

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**